ui/app/components/http-requests-bar-chart.js

```diff
@@ -0,0 +1,206 @@
+
+    const bgGradient = defs
+      .append('linearGradient')
+      .attr('id', 'bg-gradient')
+      .attr('gradientTransform', 'rotate(90)');
+
+    bgGradient
+      .append('stop')
+      // this corresponds to $blue-500
+      .attr('stop-color', '#1563ff')
+      .attr('stop-opacity', '0.8')
+      .attr('offset', '0%');
+
+    bgGradient
+      .append('stop')
+      // this corresponds to $blue-500
+      .attr('stop-color', '#1563ff')
+      .attr('stop-opacity', '0.3')
+      .attr('offset', '100%');
+
+    const clipPathRect = svgContainer
+      .append('g')
+      .attr('clip-path', `url(#clip-bar-rects)`)
+      .append('rect')
+      .attr('x', 0)
+      .attr('y', 0)
+      .style('fill', 'url(#bg-gradient)');
+
+    this.set('clipPathRect', clipPathRect);
+
+    const barsContainer = d3
+      .select('.http-requests-bar-chart')
+      .append('clipPath')
+      .attr('id', 'clip-bar-rects');
+
+    this.set('barsContainer', barsContainer);
+
+    svgContainer.append('g').attr('class', 'x axis');
+    svgContainer.append('g').attr('class', 'y axis');
+
+    this.renderBarChart();
+  },
+
+  renderBarChart() {
+    const height = this.height();
+    const { margin, width, svgContainer, xScale, yScale, parsedCounters, barsContainer, clipPathRect } = this;
+
+    // render the chart
+    d3.select('.http-requests-bar-chart')
+      .attr('width', width + margin.left + margin.right)
+      .attr('height', height + margin.top + margin.bottom)
+      .attr('viewBox', `0 0 ${width} ${height}`);
+
+    // scale and render the axes
+    const yAxis = d3Axis.axisRight(yScale).ticks(3, '.0s');
+    const xAxis = d3Axis.axisBottom(xScale).tickFormat(d3TimeFormat.timeFormat('%b %Y'));
+
+    svgContainer
+      .select('g.x.axis')
+      .attr('transform', `translate(0,${height})`)
+      .call(xAxis);
+
+    svgContainer
+      .select('g.y.axis')
+      .attr('transform', `translate(${width - margin.left - margin.right}, 0)`)
+      .call(yAxis);
+
+    // update the clipPath and render the bars
+    clipPathRect.attr('width', width).attr('height', height);
+
+    const bars = barsContainer.selectAll('.bar').data(parsedCounters);
+    const barsEnter = bars
+      .enter()
+      .append('rect')
+      .attr('class', 'bar');
+
+    bars
+      .merge(barsEnter)
+      // these attributes are only applied when bars are updated
+      .attr('width', xScale.bandwidth())
+      .attr('height', counter => height - yScale(counter.total))
+      // the offset between each bar
+      .attr('x', counter => xScale(counter.start_time))
+      .attr('y', counter => yScale(counter.total));
+  },
+
+  updateDimensions() {
+    const newWidth = this.element.clientWidth;
+    const { margin } = this;
+
+    this.set('width', newWidth - margin.left - margin.right);
+    this.renderBarChart();
+  },
+
+  waitforResize: task(function*() {
+    while (true) {
+      yield waitForEvent(window, 'resize');
+      this.updateDimensions();
```

Comment:
Do you want to add a debounce + the drop modifier here? In regular usage it's probably not necessary, but if someone's thrashing the window it'd save it until resizing was done.